audiostream/src/ni/media/audio/ieee80.h

```diff
@@ -0,0 +1,194 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+/***************************************************************\
+*   IEEE80.h							*
+*   Convert between "double" and IEEE 80 bit format  		*
+*   in machine independent manner.				*
+*   Assumes array of char is a continuous region of 8 bit frames*
+*   Assumes (unsigned long) has 32 useable bits			*
+*   billg, dpwe @media.mit.edu					*
+*   01aug91							*
+*   19aug91  aldel/dpwe  workaround top bit problem in Ultrix   *
+*                        cc's double->ulong cast		*
+*   05feb92  dpwe/billg  workaround top bit problem in 		*
+*                        THINKC4 + 68881 casting		*
+\***************************************************************/
+/* Copyright (c) 1991-1992 above authors
+Permission is hereby granted, free of charge, to any person
+obtaining a copy of this software and associated documentation
+files (the "Software"), to deal in the Software without
+restriction, including without limitation the rights to use,
+copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following
+conditions:
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+/* #define MAIN	1	 to compile test routines */
+
+#define ULPOW2TO31 ( (unsigned long) 0x80000000L )
+#define DPOW2TO31 ( (double) 2147483648.0 ) /* 2^31 */
+
+/* have to deal with ulong's 32nd bit conditionally as double<->ulong casts
+   don't work in some C compilers */
+
+static inline double        myUlongToDouble( unsigned long ul );
+static inline unsigned long myDoubleToUlong( double val );
+
+static inline double myUlongToDouble( unsigned long ul )
+{
+    double val;
+
+    /* in THINK_C, ulong -> double apparently goes via long, so can only
+       apply to 31 bit numbers.  If 32nd bit is set, explicitly add on its
+       value */
+    if ( ul & ULPOW2TO31 )
+        val = DPOW2TO31 + ( ul & ( ~ULPOW2TO31 ) );
+    else
+        val = ul;
+    return val;
+}
+
+static inline unsigned long myDoubleToUlong( double val )
+{
+    unsigned long ul;
+
+    /* cannot cast negative numbers into unsigned longs */
+    if ( val < 0 )
+    {
+        fprintf( stderr, "IEEE80:DoubleToUlong: val < 0\n" );
```

Comment:
these should probably be exceptions instead of stderr output

Reply:
true, good catch!